python/plugins/db_manager/dlg_sql_window.py

```diff
@@ -91,10 +91,8 @@ def __init__(self, iface, db, parent=None):
         self.updatePresetsCombobox()
 
         self.geomCombo.setEditable(True)
-        self.geomCombo.lineEdit().setReadOnly(True)
 
         self.uniqueCombo.setEditable(True)
-        self.uniqueCombo.lineEdit().setReadOnly(True)
```

Comment:
The impact of this modification is rather small, because it does not break anything. It's just a slight degradation of the UI if the user can provoke an error the software could have prevented beforehand.
It can be minimized further by only applying the modification when your new provider is actually used and not for the other providers. I did a similar thing when implementing composite keys in DB-Manager for the Postgres-provider, see #2382. Although the solution there is more a hack than a good piece of software engineering.

I think the bigger issue is, why can the provider not determine all column names? The way I understand the abstraction level in QGIS the provider is responsible for defining the columns and any column it can not detect is just non-existing to the rest of the system.

If you could sketch such a situation I'd be happy to have a look at the code.